src/facet_display/FacetDisplay.tsx

```diff
@@ -0,0 +1,137 @@
+import React from "react"
+import FilterableFacet from "./FilterableFacet"
+import Facet from "./Facet"
+import SearchFilter from "./SearchFilter"
+import type { FacetManifest, Facets, Aggregation, Bucket } from "./types"
+import { LEVELS, DEPARTMENTS } from "../constants"
+
+export type BucketWithLabel = Bucket & { label: string | null }
+
+interface Props {
+  facetMap: FacetManifest
+  facetOptions: (group: string) => Aggregation | null
+  activeFacets: Facets
+  onUpdateFacets: React.ChangeEventHandler<HTMLInputElement>
+  clearAllFilters: () => void
+  toggleFacet: (name: string, value: string, isEnabled: boolean) => void
```

Comment:
I think we should remove either `onUpdateFacets` or `toggleFacet`.
- both of these are for setting the value of a facet
- the way FacetDisplay is structured, if you have `toggleFacet`, you always do:
    ```tsx
    <FacetDisplay
        toggleFacet={toggleFacet}
        onUpdateFacet={e => toggleFacet(e.name, e.value, e.checked)}
        ...
    />
    ```
    This is what `useCourseSearch` returns, but IMO this component should handle it. (Either this component itself, ot Facet/FilterableFacet).

Reply:
Also: IMO, `toggleFacet` should be renamed `onFacetChange` or something. It's setting a value (via 3rd arg), not toggling a value.